control/control.go

```diff
@@ -330,13 +330,8 @@ func (c *Controller) Solve(ctx context.Context, req *controlapi.SolveRequest) (*
 	}
 
 	// if SOURCE_DATE_EPOCH is set, enable it for the exporter
-	if v, ok := epoch.ParseBuildArgs(req.FrontendAttrs); ok {
-		if _, ok := req.ExporterAttrs[epoch.KeySourceDateEpoch]; !ok {
-			if req.ExporterAttrs == nil {
-				req.ExporterAttrs = make(map[string]string)
-			}
-			req.ExporterAttrs[epoch.KeySourceDateEpoch] = v
-		}
+	if attrs, ok := epoch.SetExporterAttrFromFrontend(req.FrontendAttrs, req.ExporterAttrs); ok {
```

Comment:
Not so sure about this, but I'm guessing this is useful logic to pull into moby?

cc @tonistiigi

Reply:
I don't think we need it in moby. Just wanted to avoid having to reference `exptypes` directly in control.

Reply:
We should keep it as before if possible :+1:

It's some trickier logic that I think is specific to the control API, so I'd prefer not to hide it.